### PR TITLE
Add api acceptance tests for public webdav api actions

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add More tests for WebDav API (old and new) actions.

## Related Issue

- Related Issue #36301

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiShareOperations/changingFilesShare.feature [hunk 1]
```diff
@@ -129,4 +129,58 @@ Feature: sharing
...
+    And user "user0" has moved file "welcome.txt" to "PARENT/welcome.txt"
+    And user "user0" has created a public link share with settings
+      | path        | /PARENT   |
+      | permissions | change    |
+      | password    | newpasswd |
+    When the public deletes file "welcome.txt" from the last public share using the password "<password>" and <public-webdav-api-version> public WebDAV API
+    Then the HTTP status code should be "<http-status-code>"
+    And as "user0" file "PARENT/welcome.txt" <should-or-not> exist
+    Examples:
+      | public-webdav-api-version | http-status-code | should-or-not | password  |
```
Comment (skshetry) [inline tests/acceptance/features/apiShareOperations/changingFilesShare.feature]: It's good to be creative but, I would prefer not inventing "min" language inside scenarios.

Comment (skshetry) [inline tests/acceptance/features/apiShareOperations/changingFilesShare.feature]: It's good to be creative but, I would prefer not inventing "min" language inside scenarios.

**Comment by HariBhandari07** [comment]
LGTM

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 2]
```diff
@@ -98,3 +98,47 @@ Feature: download file
...
+    Then the downloaded content should be "example file for developers"
+
+  @public_link_share-feature-required
+  Scenario: download a public shared file with wrong password
+    Given the administrator has enabled DAV tech_preview
+    And user "user0" has created a public link share with settings
+      | path     | welcome.txt |
+      | password | newpasswd   |
+    When the public downloads the last public shared file with password "wrongpasswd" using the new public WebDAV API
+    And the value of the item "//s:message" in the response should match "/Username or password was incorrect/"
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]:
```suggestion
    Then the value of the item "//s:message" in the response should match "/Username or password was incorrect/"
```

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 3]
```diff
@@ -98,3 +98,47 @@ Feature: download file
...
+    When the public downloads the last public shared file with password "wrongpasswd" using the old public WebDAV API
+    And the value of the item "//s:message" in the response should be "Cannot authenticate over ajax calls"
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]:
```suggestion
    Then the value of the item "//s:message" in the response should be "Cannot authenticate over ajax calls"
```

Comment (phil-davis) [reply tests/acceptance/features/apiWebdavOperations/downloadFile.feature]: Do we have access to HTTP status codes here?
Do we get a 403 or?

File: tests/acceptance/features/apiShareOperations/changingFilesShare.feature [hunk 4]
```diff
@@ -129,4 +129,101 @@ Feature: sharing
     Examples:
       | public-webdav-api-version |
       | old                       |
-      | new                       |
\ No newline at end of file
+      | new                       |
+
+  @public_link_share-feature-required
+  Scenario Outline: Public can not delete file through publicly shared link with password using invalid password
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareOperations/changingFilesShare.feature]:
```suggestion
  Scenario Outline: Public cannot delete file through publicly shared link with password using invalid password
```

File: tests/acceptance/features/apiShareOperations/changingFilesShare.feature [hunk 5]
```diff
@@ -129,4 +129,101 @@ Feature: sharing
...
+    When the public deletes file "welcome.txt" from the last public share using the password "newpasswd" and <public-webdav-api-version> public WebDAV API
+    Then the HTTP status code should be "204"
+    And as "user0" file "PARENT/welcome.txt" should not exist
+    Examples:
+      | public-webdav-api-version |
+      | old                       |
+      | new                       |
+
+  @public_link_share-feature-required
+  Scenario Outline: Public tries to rename a file in a password protected share using the invalid password
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareOperations/changingFilesShare.feature]:
```suggestion
  Scenario Outline: Public tries to rename a file in a password protected share using an invalid password
```

File: tests/acceptance/features/apiShareOperations/changingFilesShare.feature [hunk 6]
```diff
@@ -129,4 +129,101 @@ Feature: sharing
...
+    When the public uploads file "lorem.txt" with password "invalid" and content "test" using the <public-webdav-api-version> public WebDAV API
+    Then the HTTP status code should be "401"
+    And as "user0" file "/PARENT/lorem.txt" should not exist
+    Examples:
+      | public-webdav-api-version |
+      | old                       |
+      | new                       |
+
+  @public_link_share-feature-required
+  Scenario Outline: Public tries to upload to a password protected public share using an valid password
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareOperations/changingFilesShare.feature]:
```suggestion
  Scenario Outline: Public tries to upload to a password protected public share using the valid password
```

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 7]
```diff
@@ -98,3 +98,47 @@ Feature: download file
...
+
+  @public_link_share-feature-required
+  Scenario: download a public shared file without password
+    Given the administrator has enabled DAV tech_preview
+    And user "user0" has created a public link share with settings
+      | path     | welcome.txt |
+      | password | newpasswd   |
+    When the public downloads the last public shared file using the new public WebDAV API
+    And the value of the item "//s:message" in the response should match "/No 'Authorization: Basic' header found/"
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]:
```suggestion
    Then the value of the item "//s:message" in the response should match "/No 'Authorization: Basic' header found/"
```

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 8]
```diff
@@ -98,3 +98,47 @@ Feature: download file
...
+    When the public downloads the last public shared file with password "" using the old public WebDAV API
+    And the value of the item "//s:message" in the response should be "Cannot authenticate over ajax calls"
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]:
```suggestion
    Then the value of the item "//s:message" in the response should be "Cannot authenticate over ajax calls"
```

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 9]
```diff
@@ -98,3 +98,47 @@ Feature: download file
...
+
+  @public_link_share-feature-required
+  Scenario: try to download the public share with upload only permessions
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]:
```suggestion
  Scenario: try to download from a public share that has upload only permissions
```

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 10]
```diff
@@ -98,3 +98,47 @@ Feature: download file
...
+    Given the administrator has enabled DAV tech_preview
+    And user "user0" has created a public link share with settings
+      | path        | PARENT          |
+      | permissions | uploadwriteonly |
+    When the public downloads file "parent.txt" from inside the last public shared folder using the new public WebDAV API
+    Then the downloaded content should be "ownCloud test text file parent" plus end-of-line
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]: I am surprised. The download should not work???

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 11]
```diff
@@ -98,3 +98,47 @@ Feature: download file
...
+    When the public downloads file "parent.txt" from inside the last public shared folder using the old public WebDAV API
+    Then the downloaded content should be "ownCloud test text file parent" plus end-of-line
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]: I am surprised. The download should not work???

File: tests/acceptance/features/apiWebdavOperations/publicWebDAVAPIDisabled.feature [hunk 12]
```diff
@@ -48,3 +48,17 @@ Feature: the new public WebDAV API is not available when the tech preview settin
     When the public uploads file "lorem.txt" with content "test" using the new public WebDAV API
     Then the HTTP status code should be "401"
     And as "user0" file "/PARENT/lorem.txt" should not exist
+
+  @public_link_share-feature-required
+  Scenario: Public cannot upload file after the public WebDAV API when it is disabled
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/publicWebDAVAPIDisabled.feature]:
```suggestion
  Scenario: Public cannot upload file when the public WebDAV API is disabled
```

**Review by phil-davis — COMMENTED** [review]
Mostly just small words to change. But:
- there is one surprising expected test result (see review comments)
- can we check status codes in a few more places for scenarios where the operation should not work

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature (hunk 3 above)
Comment (dpakach) [reply tests/acceptance/features/apiWebdavOperations/downloadFile.feature]: Yes we do. The status code is 401. I am adding the check for status code currently.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**